share/completions/pipenv.fish

```diff
@@ -0,0 +1,3 @@
+if test (command -v pipenv)
+    eval (pipenv --completion)
```

Comment:
Are there any guarantees that `--completion` will always be fish-compatible?

Reply:
It is guarantee to always return completion for the shell pointed by the `SHELL` environment variable. This is the way is documented in the official documentation and I checked the source code (of Pipenv) as well.

Reply:
It's certainly producing fish-compatible output intended to be `eval`'d. 

```complete --command pipenv --arguments "(env _PIPENV_COMPLETE=complete-fish COMMANDLINE=(commandline -cp) pipenv)" -f`

Reply:
Yeah, I checked the source code for [pipenv](https://github.com/kennethreitz/pipenv) and [click-completion](https://github.com/click-contrib/click-completion) before submitting the patch. Pretty good code if you ask me.

The only concern will be if you set `SHELL` as something that is not fish, but I think in that case not only this completion will fail.